SECURITY.WHITEPAPER.md

```diff
@@ -0,0 +1,113 @@
+Default user creation process consists of creation private profile, public profile, keystore with public keys, keystore with private keys.
+```
+Default location within system dfs:
+    /profiles
+        /private
+            /username - user's private profile 
+        /public
+            /username - user's public profile   
+    /users
+        /public
+            /pubkeys - keystore consists user's public key
+            /inbox - location of shared with user files    
+        /private
+            /keystore - keystore consists user's private key
+            /files/SIV - location of private files. SIV is 3 symbol path encryption algorithm identifier.
+```                   
+Example private profile:
+```
+    {
+        "keystore": {
+            "resource": "s3://adorsys-docusafe/users/username/private/keystore"
+        },
+        "privateStorage": [
+            [{"id": "DEFAULT"}, {"resource": "s3://adorsys-docusafe/users/username/private/files/"}]
+        ],
+        "inboxWithFullAccess": {
+            "resource": "s3://adorsys-docusafe/users/username/public/inbox/"
+        },
+        "publishPublicKeysTo": {
+            "resource": "s3://adorsys-docusafe/users/username/public/pubkeys"
+        },
+        "associatedResources": [
+            {"resource": "s3://adorsys-docusafe/users/username/"}
+        ],
+        "documentVersionStorage": {
+            "resource": "s3://adorsys-docusafe/users/username/versions/"
+        },
+        "appVersion": "BASELINE"
+    }
+```
+            
+Example public profile:
+```
+    {
+        "publicKeys": {
+            "resource": "s3://adorsys-docusafe/users/username/public/pubkeys"
+        },
+        "inbox": {
+            "resource": "s3://adorsys-docusafe/users/username/public/inbox/"
+        },
+        "appVersion": "BASELINE"
+    }   
+```
+    
+## Keystore encryption
+User personal password is used to access keystore. Password can be changed without the need of changing keystore content.
+By default BCFKS keystore type is used. Keystore encryption algorithm is AES256_KWP. Password key derivation algorithm 
+PRF_SHA512 with salt length 32 and iteration count 20480, KeyStore authentication algorithm HmacSHA3_512, password-like 
+keys encryption algorithm is PBEWithHmacSHA256AndAES_256.
+All this parameters can be changed by setting keystore config. For example instead of BCFKS keystore can be used UBER, 
+instead of PBKDF2 based routines can be used Scrypt based.
+Keystore contains secret keys for private files and path encryption, public/private key pairs for sharing files and 
+certificates for sign/verify files. 
+Default keystore prefix for generated keys is "KEYSTORE-ID-0". Each signing and encryption key has unique alias which 
+is formed from keystore prefix plus UUID.
+
+## Private files encryption algoritm 
+Datasafe files uploaded by users in private area are encrypted using symmetric key 256-bit Advanced Encryption Standard (AES).
```

Comment:
AES GCM is used for inbox too - the only difference between INBOX/PRIVATE is content encryption key derivation algorithm